xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/js/xwiki/xwiki.js

```diff
@@ -253,8 +253,8 @@ Object.extend(XWiki, {
                       });
 
                       // Replace the element's content with the temporary container's content, while also evaluating any inline scripts.
-                      // Note: This also does stripScripts internally, but it's OK now, since we have already added the necessary external scripts to head.
-                      $(extraID + "pane").update(container.innerHTML);
+                      // Note: This also allows script tags defined in the updated html to be loaded.
+                      $(extraID + "pane").update(container.descendants()[0]);
```

Comment:
This usage of `.descendants()[0]` need to be explained. I'm not sure to understand it in the right context, so I'm pretty sure I won't understand it in 2 weeks.

Reply:
The script is now using `childElements`. I found out at the last minute that the `descendant()[0]` was breaking the history pane...
See https://github.com/xwiki/xwiki-platform/pull/1315/files#diff-3c136b7fe6897859a417c7b58211182dR255